plugin/nfs/nfs.py

```
@@ -0,0 +1,388 @@
+            raise LsmError(ErrorNumber.INVALID_ARGUMENT,
+                           'Must provide fs_id or export_path')
+
+        common_opts = []
+        if options is not None:
+            common_opts.extend(options.split(','))
+
+        if anon_uid is not None and anon_uid != NfsExport.ANON_UID_GID_NA:
+            common_opts.append('anonuid=%d' % anon_uid)
+
+        if anon_gid is not None and anon_gid != NfsExport.ANON_UID_GID_NA:
+            common_opts.append('anongid=%d' % anon_gid)
+
+        if auth_type is not None:
+            common_opts.append('sec=%s' % str(auth_type))
+
+        if export_path is None:
+            export_path = NFSPlugin._get_fsid_path(fs_id)
+            if export_path is None:
+                raise LsmError(ErrorNumber.INVALID_ARGUMENT,
+                               'Could not locate filesystem')
+
+        for host in rw_list:
+            opts = copy.copy(common_opts)
+            if host in root_list:
+                opts.append('no_root_squash')
+            opts.append('rw')
+            try:
+                NFSPlugin._add_export(export_path, host, opts)
+            except Exception:
+                raise
+
+        for host in ro_list:
+            opts = copy.copy(common_opts)
+            if host in root_list:
+                opts.append('no_root_squash')
+            opts.append('ro')
+            try:
+                NFSPlugin._add_export(export_path, host, opts)
+            except Exception:
+                raise
+
+        NFSPlugin._update_exports()
+
+    def export_remove(self, export, flags=0):
+        try:
+            exports = NFSPlugin._load_exports()
+            filtered = NFSPlugin._filter_export(exports, export.id)
+            NFSPlugin._write_exports(filtered)
+            NFSPlugin._update_exports()
+        except Exception:
+            raise
+
+    def plugin_unregister(self, flags=0):
+        return
+
+    def job_status(self, job_id, flags=0):
+        return
```

@jumitche There is no `job_create`.  Jobs are created as side effects of API calls that don't finish before they return.  If a plugin has no async. functionality that returns jobs then these functions become stubs that don't do anything.  I would just raise an Exception as a client shouldn't be calling these methods if the plugin is not returning a job(s).